test/extended/workspacetype/workspace.go

```diff
@@ -16,49 +18,90 @@ var _ = g.Describe("[sig-workspace]", func() {
+		g.By("# Create five child workspaces under the parent workspace")
+		childWorkSpaces := []exutil.WorkSpace{}
+		for i := 1; i <= 5; i++ {
+			g.By(fmt.Sprintf("# Creating child workspace no.%v", i))
+			k.SetupWorkSpaceWithSpecificPath(parentWorkSpace.ServerURL)
+			childWorkSpace := k.WorkSpace()
+			childWorkSpaces = append(childWorkSpaces, childWorkSpace)
+		}
 
-		g.By("# From the home workspace could get the test workspace but couldn't get its child level workspace")
-		output, err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+myWorkSpace.ParentServerURL, "workspace").Output()
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(output).Should(o.And(
-			o.ContainSubstring(myWorkSpace.Name),
-			o.ContainSubstring(myWorkSpace.ServerURL),
-		))
-		o.Expect(output).ShouldNot(o.ContainSubstring(mySubWorkSpace.Name))
-
-		g.By("# From the test workspace could get its child workspace")
-		output, err = k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+myWorkSpace.ServerURL, "workspace").Output()
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(output).Should(o.And(
-			o.ContainSubstring(mySubWorkSpace.Name),
-			o.ContainSubstring(mySubWorkSpace.ServerURL),
-		))
-
-		g.By("# Delete the child workspace and the test workspace should be successful")
-		// Delete the child workspace and check it deleted successfully
-		output, err = k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+myWorkSpace.ServerURL, "workspace", mySubWorkSpace.Name).Output()
+		g.By("# From the home workspace could get parent workspace but couldn't get child workspaces")
+		homeSubWorkspaces := k.ListWorkSpacesWithSpecificPath(parentWorkSpace.ParentServerURL)
+		// NOTE: use contain substring as a workaround to WorkSpace name inequality bug: https://url.corp.redhat.com/1878a60
+		homeSubWorkspacesString := strings.Join(homeSubWorkspaces, ",")
+		o.Expect(homeSubWorkspacesString).Should(o.ContainSubstring(parentWorkSpace.Name))
+		for _, childWorkspace := range childWorkSpaces {
+			o.Expect(homeSubWorkspacesString).ShouldNot(o.ContainSubstring(childWorkspace.Name))
+		}
+
+		g.By("# From parent workspace could get its child workspace")
+		parentSubWorkspaces := k.ListWorkSpacesWithSpecificPath(parentWorkSpace.ServerURL)
+		// NOTE: use contain substring as a workaround to WorkSpace name inequality bug: https://url.corp.redhat.com/1878a60
+		parentSubWorkspacesString := strings.Join(parentSubWorkspaces, ",")
+		for _, childWorkspace := range childWorkSpaces {
+			o.Expect(parentSubWorkspacesString).Should(o.ContainSubstring(childWorkspace.Name))
+		}
+
+		g.By("# Delete child workspaces")
+		for _, childWorkspace := range childWorkSpaces {
+			output, err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+childWorkspace.ParentServerURL, "workspace", childWorkspace.Name).Output()
```

Comment:
@Phaow  #7 cc'ed you. If you took a look, you could notice WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer is not needed any longer after
his PR got merged.

Please update other places in your PR as well.

Reply:
Sure. Thanks! I'll enhance it next PR.

Reply:
@Phaow, I mean above comment was for @zimo-xiao : ). Not for you. Updated comment clearer.

Reply:
@zimo-xiao you need rebase latest code for this.

Reply:
> @Phaow, I mean above comment was for @zimo-xiao : ). Not for you. Updated comment clearer.

Got it. Thanks for clarify!